app/src/main/java/dev/baseio/discordjetpackcompose/ui/routes/onboarding/commonui/DiscordDialog.kt

```diff
@@ -0,0 +1,132 @@
+import androidx.compose.runtime.Composable
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.graphics.Color
+import androidx.compose.ui.graphics.RectangleShape
+import androidx.compose.ui.graphics.Shape
+import androidx.compose.ui.res.stringResource
+import androidx.compose.ui.text.TextStyle
+import androidx.compose.ui.text.font.FontWeight
+import androidx.compose.ui.unit.dp
+import androidx.compose.ui.unit.sp
+import androidx.compose.ui.window.Dialog
+import androidx.compose.ui.window.DialogProperties
+import dev.baseio.discordjetpackcompose.R
+import dev.baseio.discordjetpackcompose.ui.theme.DiscordColorProvider
+import dev.baseio.discordjetpackcompose.ui.theme.black
+import dev.baseio.discordjetpackcompose.ui.theme.white
+
+@Composable
+fun DiscordDialog(
+    modifier: Modifier = Modifier,
+    show: Boolean,
+    shape: Shape = RectangleShape,
+    backgroundColor: Color = white,
+    buttonRowBackgroundColor: Color = Color.LightGray,
+    titleTextProvider: () -> Int,
+    subTitleTextProvider: () -> Int,
+    confirmActionButtonTextProvider: () -> Int,
+    properties: DialogProperties = DialogProperties(),
+    onDismissRequest: () -> Unit,
+    onClickCancelButton: () -> Unit,
+    onClickConfirmActionButton: () -> Unit
+) {
+    if (show) {
+        Dialog(
+            onDismissRequest = { onDismissRequest() },
+            properties = properties
+        ) {
+            Surface(
+                modifier = modifier,
+                shape = shape,
+                color = backgroundColor
+            ) {
+                Column(
+                    modifier = Modifier.padding(top = 16.dp)
+                ) {
+                    Text(
+                        text = stringResource(id = titleTextProvider()),
+                        style = TextStyle(
+                            fontWeight = FontWeight.ExtraBold,
+                            fontSize = 20.sp,
+                            color = black
+                        ),
+                        modifier = Modifier
+                            .padding(bottom = 16.dp, top = 10.dp)
+                            .padding(horizontal = 16.dp)
+                    )
+                    Divider(color = Color.LightGray, thickness = 0.5.dp)
+                    Text(
+                        text = stringResource(id = subTitleTextProvider()),
+                        style = TextStyle(
+                            fontWeight = FontWeight.Normal,
+                            fontSize = 16.sp,
+                            color = black
+                        ),
+                        modifier = Modifier.padding(
+                            top = 12.dp,
+                            start = 16.dp,
+                            end = 16.dp,
+                            bottom = 24.dp
+                        )
+                    )
+                    Divider(color = Color.LightGray, thickness = 0.5.dp)
```

Comment:
@yugeshjain saw this in another PR also, make sure we resolve the comments from there